overlay-ot.conf

```diff
@@ -0,0 +1,49 @@
+# Hack: Disable BT on devices which support both for now
+CONFIG_BT=n
+
+CONFIG_CPLUSPLUS=y
+CONFIG_FLASH=y
```

Comment:
Why are some options given in this overlay with the same values they have in prj.conf? That shouldn't be necessary.

For options which need different options when using IPSP vs. OpenThread, does it make sense to handle in the Kconfig using default values to have a single point of truth instead of relying on kconfig to override prj.conf with overlay-ot.conf values? Seems a little nicer to me and may result in a smaller overlay file which is easier to read.